lib/collection/src/collection/mod.rs

```diff
@@ -532,6 +513,14 @@ impl Collection {
             // Try to find dead replicas with no active transfers
             let transfers = shard_holder.get_transfers(|_| true);
 
+            // Respect shard transfer limit, consider already proposed transfers in our counts
+            let (mut incoming, outgoing) = shard_holder.count_shard_transfer_io(this_peer_id);
+            incoming += proposed.get(this_peer_id).copied().unwrap_or(0);
+            if self.check_shard_transfer_limit(incoming, outgoing) {
+                log::trace!("Postponing automatic shard {shard_id} transfer to stay below limit on this node (i/o: {incoming}/{outgoing})");
```

Comment:
Final suggestion:

```suggestion
                log::trace!("Postponing automatic shard {shard_id} transfer to stay below limit on this node (incoming: {incoming}, outgoing: {outgoing})");
```

Reply:
I had this originally but changed it to `i/o` because I thought it was quite long.

But happy to change it back :+1: 

Thanks for another review round!